src/Analyzer.TemplateProcessor/ArmTemplateProcessor.cs

```diff
@@ -182,6 +173,133 @@ internal Template ProcessTemplateResourceAndOutputPropertiesLanguageExpressions(
+        /// <param name="parentType">Type of the parent resource. Used during recursive call.</param>
+        /// <returns>Dictionary that maps full resource names and types to their corresponding resource.</returns>
+        private Dictionary<string, TemplateResource> FlattenResources(TemplateResource[] resources, string parentName = null, string parentType = null)
+        {
+            Dictionary<string, TemplateResource > flattenedResources = new Dictionary<string, TemplateResource>(StringComparer.OrdinalIgnoreCase);
+
+            foreach (var resource in resources)
+            {
+                string dictionaryKey;
+
+                if (parentName != null && parentType != null)
+                {
+                    dictionaryKey = $"{parentName}/{resource.Name.Value} {parentType}/{resource.Type.Value}";
+                }
+                else
+                {
+                    dictionaryKey = $"{resource.Name.Value} {resource.Type.Value}";
+                }
+
+                flattenedResources.Add(dictionaryKey, resource);
+
+                if (resource.Resources != null)
+                {
+                    flattenedResources.AddRange(FlattenResources(resource.Resources, resource.Name.Value, resource.Type.Value));
+                }
+            }
+
+            return flattenedResources;
+        }
+
+        /// <summary>
+        /// Processes language expressions in the properties property of the resources.
+        /// </summary>
+        /// <param name="templateResource">The template resource to process language expressions for.</param>
+        /// <param name="evaluationHelper">Evaluation helper to evaluate expressions</param>
+        /// <returns>The parsed template as a Template object.</returns>
+        private TemplateResource ProcessTemplateResourceLanguageExpressions(TemplateResource templateResource, TemplateExpressionEvaluationHelper evaluationHelper)
+        {
+            try
+            {
+                if (templateResource.Properties != null)
+                {
+                    evaluationHelper.OnGetCopyContext = () => templateResource.CopyContext;
+                    templateResource.Properties.Value = ExpressionsEngine.EvaluateLanguageExpressionsRecursive(
+                        root: templateResource.Properties.Value,
+                        evaluationContext: evaluationHelper.EvaluationContext);
+                }
+
+                if (templateResource.Resources != null)
+                {
+                    foreach (var childResource in templateResource.Resources)
+                    {
+                        ProcessTemplateResourceLanguageExpressions(childResource, evaluationHelper);
+                    }
+                }
+            }
+            catch
+            {
+                // Do not throw if there was an issue with evaluating language expressions
+                return null;
```

Comment:
I don't see usages of the return values of this function

Reply:
True. I can make it void